packages/web/src/common/utils/uid.ts

```diff
@@ -73,40 +87,47 @@ export function makeUids(kinds, ids, source) {
   }
 }
 
-export function makeUid(kind, id, source) {
+export function makeUid(kind: string, id: ID, source?: string) {
   const uid = Uid.build(kind, id, source)
   return uid.toString()
 }
 
 /**
  * A persistant identifier for a resource for re-use within a session.
  * Differs from a uid, which represents an instance.
- * @param {Kind} kind
- * @param {ID} id
+ * @param kind
+ * @param id
  */
-export const makeKindId = (kind, id) => `${kind}-${id}`
+export const makeKindId = (kind: string, id: ID | string | undefined) =>
+  `${kind}-${id}`
 
 /**
  * Gets the id from a kind id
- * @param {String} kindId
+ * @param kindId
  */
-export const getIdFromKindId = kindId => kindId.split('-')[1]
+export const getIdFromKindId = (kindId: string) => kindId.split('-')[1]
 
 /**
  * Gets the kind from a kind id
- * @param {String} kindId
+ * @param kindId
  */
-export const getKindFromKindId = kindId => kindId.split('-')[0]
+export const getKindFromKindId = (kindId: string) => kindId.split('-')[0]
 
 export const uuid = () => {
```

Comment:
maybe we can just use a lib for this instead of typing our own

Reply:
def down for that, typing here helped me see all the callers that were making assumptions etc